docs/version-locking.md

```diff
@@ -0,0 +1,124 @@
+
+Ultimately, redering a specific set of services is initiated by the VCL via strains. By selecting the specific strain, the respective `x-ow-version-lock` header is sent to the respective `helix-dispatch` service. The strain is selected, either via the helix pages branch logic, or a strain condition, or a pre-flight condition.
+
+The _deploy_ script in the helix-pages repository ensures that the `versions.yaml` of the respective branch is properly transformed into a _strain_ in the `helix-config.yaml`.
+If the `versions.yaml` also contains an entry for the `pipeline`, it updates the `package.json`. 
+
+(the publishing step will update the respective edge dictionary for that strain).
+
+### Missing features
+
+- [ ] The `ow-x-version-lock` header must be strain dependant
+- [ ] The `dispatch` version must be strain dependant
+
+## Selecting the (default) version set
+
+As long as the project is _surfed_ on the specific version branch (eg `https://breaking-december--pages--adobe.com/`), the branch name automatically selects the version set (via the strain). But once the feature branch is merged back to `main`, this information is lost. 
+
+Using preflight requests and helix pages version picker service allow to define additional information that can be used in strain conditions:
+
+The default branch of a helix pages project might have a `helix-version.txt` file which contains the name of a version set (eg. `breaking-december`). The version picker service will respond with a `x-helix-version` header, containing the name of the version set. eg: `x-helix-version: breaking-december`.
+
+The preflight request login in the VCL will set the `preflight.x-helix-version` condition which will select the correct strain.
+
+## Implications
+
+### Locked-in problem
+
+Once a project chooses a specific version, eg. the `helix-version.txt` contains the `breaking-december`, helix needs to ensure that the respective version set remains active and once removed, the default version set needs to be compatible with the version set.
+
+Example:
+
+1. helix creates the `breaking-december` version set that contains backward incompatible changes.
+2. project creates a `breaking-december` branch and (automatically) a `helix-versions.txt` (containing `breaking-december`). 
+3. project updates its client code until the breaking issues are fixed.
+4. project merges its `breaking-december` branch back to `main`, which now contains the `helix-versions.txt` with the `breaking-december` version name.
+5. helix creates the `breaking-january` version set
+6. project **doesn't update** to breaking-january
+7. helix creates the `breaking-february` version set
+8. `breaking-december` will now be the new default and all services are updated accordingly.
+
+And so on... As long a 1 project still contains a reference to an old _version_, the strain in helix-pages needs to remain. The risk here is to reach the limit of the edge dictionaries or to loose oversight.
```

Comment:
```suggestion
And so on… As long a 1 project still contains a reference to an old _version_, the strain in helix-pages needs to remain. The risk here is to reach the limit of the edge dictionaries or to loose oversight.
```

Reply:
I think a bigger risk is the customers who want to stay perpetually on an old version, but also have new features. Therefore I'd treat `version.txt` as an indication of preference, and not as a hard choice.